module/VuFind/src/VuFind/Auth/MultiILS.php

```diff
@@ -123,4 +124,51 @@ public function setCatalog(\VuFind\ILS\Connection $connection)
         }
         parent::setCatalog($connection);
     }
+
+    /**
+     * Does this authentication method support password recovery
+     *
+     * @param ?string $target Authentication target for methods that support target selection
+     *
+     * @return bool
+     *
+     * @throws \Exception
+     */
+    public function supportsPasswordRecovery(?string $target = null)
+    {
+        if (!$target) {
+            throw new \Exception(__METHOD__ . ' requires the target parameter!');
+        }
+        $recoveryConfig = $this->getCatalog()->checkFunction(
+            'resetPassword',
+            ['cat_username' => "$target.123"]
+        );
+        return $recoveryConfig ? true : false;
+    }
+
+    /**
+     * Get password recovery data (such as a user id or recovery token) based on form data submitted by the user.
+     *
+     * @param array $params Request params (form data)
+     *
+     * @return ?array Null if user not found, or associative array with following keys:
+     *   string email    User's email address
+     *   string username Username (optional, for display)
+     *   array  details  Array of user details required for resetPassword request
+     */
+    public function getPasswordRecoveryData(array $params): ?array
+    {
+        if (!($target = $params['target'] ?? null)) {
+            throw new \Exception(__METHOD__ . ' requires the target parameter!');
+        }
+        $params['cat_username'] = $target . '.' . $params['cat_username'];
+
+        $result = $this->getCatalog()->getPasswordRecoveryData($params);
+        if (!$result['success']) {
+            throw new AuthException($result['error']);
```

Comment:
Should we put a fallback here just in case?

```suggestion
            throw new AuthException($result['error'] ?? 'authentication_error_technical');
```